CONTRIBUTING.md

```diff
@@ -0,0 +1,46 @@
+2. We're here to help each other learn, after all. Don't hesitate to talk to us at any point! [The Fabric Discord](https://discord.gg/v6v4pMv) is a great place to ask questions and get input on your contributions.
+3. Fabric isn't easy to get your content into, as we uphold standards of quality and are very paranoid about making a bad precedent. You will be scrutinized again, and again, and again; you will be asked questions about more-or-less every line in your pull request (even the imports!).
+
+## Knowing When To Aim For Fabric
+
+The first step to contributing to Fabric API is coming up with an idea, and while all ideas may be useful, not all ideas necessarily fit within the scope of Fabric API.
+
+Fabric API tends to be restricted to small, highly used, and breaking changes to the vanilla game. Examples include helpers for adding blocks, which are used by basically every modder ever (hence their inclusion); breaking changes to the rendering system, which ensure that mods can remain compatible with one another; or commonly used mixins that, because of their nature, would otherwise cause mods to be incompatible (think redirects or overwrites). At the end of the day Fabric API exists for ease of implementing and modifying vanilla features, and compatibility between mods.
+
+If your contribution is a massive system, or api for something that doesn't exist in the vanilla game, it does not belong in Fabric API. The best examples are things like, custom world generators, mana systems, or a fluid api. All of these things represent content expansions that don't exist in the base game and should not be included in Fabric API. That doesn't mean however, that your idea is not useful. Larger, more experimental libraries are being developed over at [FabLabs](https://github.com/FabLabsMC). They might be interested in your idea.
+
+If you're unsure of whether or not your idea fits within the scope of Fabric API, do it anyways. Worst case scenario Fabric API denies your pull request, but the community get's a really cool library that does something useful, best case scenario your project is exactly what we're looking for Fabric API. If you don't want to waste your time, however, ask on the Fabric Discord and get the opinion of other members of the community to see if they would appreciate the contribution.
+
+## Taking The First Step
+
+The first step to contributing to Fabric API is making an issue. Head over to the [Fabric API GitHub repository](https://github.com/FabricMC/fabric/issues/) and create a detailed issue outlining what you'd like added to Fabric API. From there, the community can weigh in on whether or not that idea is within the scope of Fabric API, how to best implement the idea, and any additional details that need ironing out.
```

Comment:
I wouldn't call it required, but a little closer to "You are encouraged to make an issue".